src/search.jl

```diff
@@ -544,3 +546,410 @@ function execute(search::GeneticAlgorithmSearch, problem)
         end
     end
 end
+
+
+#=
+Julia - GSOC
+
+Ahmed Madbouly
+
+Implementation for AND-OR, Online DFS, LRTA
+=#
+
+abstract type AbstractProblemStruct end;
+
+mutable struct ProblemStruct <: AbstractProblemStruct
+    initial::String
+    goal::Nullable{String}
```

Comment:
Should goal not be of type `State`? Why it's a Nullable string? So is initial? There is already `AbstractProblem` in the project. What advantage does `AbstractProblemStruct` bring in?